### PR TITLE
LPS-144175 Convert ImportObjectDefinitionModal to typescript

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-144175

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Review by miltonmc — CHANGES_REQUESTED** [review]
I found the following minor issues in the conversion to TypeScript:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx [hunk 1]
```diff
@@ -16,22 +16,33 @@ import ClayAlert from '@clayui/alert';
...
+}
+
+const ImportObjectDefinitionModal: React.FC<IProps> = ({
 	importObjectDefinitionURL,
 	nameMaxLength,
 	portletNamespace,
 }) => {
 	const [visible, setVisible] = useState(false);
-	const inputFileRef = useRef();
+	const inputFileRef = useRef<any>();
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx:39]: Hi @miltonmc, for some reason, with the `HTMLInputElement` type this problem is generated:

```
Type 'MutableRefObject<HTMLInputElement | undefined>' is not assignable to type 'LegacyRef<HTMLInputElement> | undefined'.
  Type 'MutableRefObject<HTMLInputElement | undefined>' is not assignable to type 'RefObject<HTMLInputElement>'.
    Types of property 'current' are incompatible.
      Type 'HTMLInputElement | undefined' is not assignable to type 'HTMLInputElement | null'.
        Type 'undefined' is not assignable to type 'HTMLInputElement | null'.ts(2322)
index.d.ts(100, 9): The expected type comes from property 'ref' which is declared here on type 'DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>'
```
So I find this another type that we can use:

```js 
const inputFileRef = useRef() as React.MutableRefObject<HTMLInputElement>;
```

What do you think?

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx [hunk 2]
```diff
@@ -149,12 +163,15 @@ const ImportObjectDefinitionModal = ({
 						className="d-none"
 						name={objectDefinitionJSONInputId}
 						onChange={({target}) => {
-							const [inputFile] = target.files;
+							const inputFile = target.files?.item(0);
+							console.log(inputFile);
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx:167]: Ops, my mistake :rofl:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx [hunk 3]
```diff
@@ -149,12 +163,15 @@ const ImportObjectDefinitionModal = ({
...
+
 							setFile({
-								fileName: inputFile.name,
+								fileName: inputFile?.name,
 								inputFile,
 								inputFileValue: target.value,
 							});
+							console.log(inputFile, fileName);
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx:174]: Again, my mistake... :man_facepalming:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx [hunk 4]
```diff
@@ -66,13 +77,16 @@ const ImportObjectDefinitionModal = ({
...
 			<ClayModal.Header>
 				{Liferay.Language.get('import-object')}
 			</ClayModal.Header>
 
 			<ClayModal.Body>
 				<ClayForm
+
+					// @ts-ignore
+
 					action={importObjectDefinitionURL}
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/modals/ImportObjectDefinitionModal.tsx:90]: Hey @miltonmc, I think we've talked about this issue, and it looks like it's something related to ClayForm types, this action property keeps giving error without `@ts-ignore` :thinking:

**Pull request closed without merging**